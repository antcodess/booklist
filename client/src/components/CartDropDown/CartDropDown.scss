.cart-dropdown {
  background-color: white;
  position: absolute;
  width: 240px;
  height: 340px;
  display: flex;
  flex-direction: column;
  border-radius: 5px;
  box-shadow: 2px 1px 10px #c3c3c3;
  top: 110%;
  right: 0%;
  z-index: 5;

  .cart-items {
    height: 250px;
    margin-left: 10px;
    flex-direction: column;
    overflow: scroll;

    .cart-message {
      display: flex;
      font-family: 'Cera GR Bold';
      color: #c3c3c3;
      margin: 50% auto 0;
    }
  }

  .custom-button {
    width: 90%;
    margin: auto;
  }
}
