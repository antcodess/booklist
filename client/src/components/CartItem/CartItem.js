import React from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

import { addItem, removeItem } from '../../redux/cart/cart.actions';

import './CartItem.scss';

export const CartItem = ({ item, history, addItem, removeItem }) => {
  const { title, imageUrl, price, quantity } = item;

  return (
    <div className='cart-item'>
      <img src={imageUrl} alt='item' onClick={() => {
        history.push(`/${item.routeName}`)
      }}/>
      <div className='item-details'>
        <span className='name' onClick={() => {
          history.push(`/${item.routeName}`)
        }}>{title}</span>
        <span className='quantity'>
        <div className='arrow' id='removeItemArrow' onClick={() => removeItem(item)}>&#10094;</div>
        <span className='value'>{quantity} x ${price}</span>
        <div className='arrow' id='addItemArrow' onClick={() => addItem(item)}>&#10095;</div>
      </span>
      </div>
    </div>
  );
};

export const mapDispatchToProps = dispatch => ({
  addItem: item => dispatch(addItem(item)),
  removeItem: item => dispatch(removeItem(item))
});

export default connect(null, mapDispatchToProps)(withRouter(CartItem));

