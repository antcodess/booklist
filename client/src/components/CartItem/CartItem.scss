.cart-item {
  width: 100%;
  display: flex;
  height: 70px;
  margin-top: 15px;

  img {
    width: 30%;
  }
  
  .item-details {
    width: 68%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;

    .name {
      font-family: "Cera GR Bold";
      font-size: 1rem;
    }
    
    .quantity {
      display: flex;
      justify-content: center;
      margin-top: 2px;
  
      .arrow {
        cursor: pointer;
      }
  
      .value {
        margin: 2px 10px;
        font-size: 0.94rem;
      }
    }
  }
}