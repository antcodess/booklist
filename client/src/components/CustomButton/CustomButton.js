import React from 'react';

import './CustomButton.scss';

const CustomButton = ({ children, isGoogleSignIn, isRegularSignIn, ...otherProps }) => {
  return (
    <button 
    className={`custom-button ${isGoogleSignIn ? 'google-sign-in' : ''}${isRegularSignIn ? 'regular-sign-in' : ''}`} 
    {...otherProps}>{ children }</button>
  );
};

export default CustomButton;