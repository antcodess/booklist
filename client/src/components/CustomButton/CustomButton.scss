.regular-sign-in {
  margin-bottom: 5px;
  &:hover {
    background-color: black;
    opacity: 0.9;
  }
}

.google-sign-in {
  background-color: #0080DE;

  &:hover {
    opacity: 0.9;
  }
}

@media only screen and (min-width: 950px) {
  .regular-sign-in {
    width: 40%;
  }
}

@media only screen and (min-width: 1100px) {
  .regular-sign-in, .google-sign-in {
    width: 49%;
  }
}