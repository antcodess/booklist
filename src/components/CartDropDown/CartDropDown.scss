.cart-dropdown {
  background-color: white;
  position: absolute;
  width: 240px;
  height: 340px;
  display: flex;
  flex-direction: column;
  border-radius: 5px;
  box-shadow: 2px 1px 10px #c3c3c3;
  top: 110%;
  right: 0%;
  z-index: 5;
  
  .cart-items {
    height: 250px;
    margin-left: 10px;
    display: flex;
    flex-direction: column;
    overflow: scroll;
    
    .cart-message {
      font-family: "Cera GR Bold";
      color: #c3c3c3;
      margin: auto;
    }
  }

  .custom-button {
    width: 90%;
    margin: auto;
  }
}

