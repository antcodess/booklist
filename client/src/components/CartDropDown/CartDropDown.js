import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { withRouter } from 'react-router-dom';

import CartItem from '../CartItem/CartItem';
import CustomButton from '../CustomButton/CustomButton';

import { toggleCartHidden } from '../../redux/cart/cart.actions';
import { selectCartItems } from '../../redux/cart/cart.selectors';

import './CartDropDown.scss';

const CartDropDown = ({ cartItems, history, toggleCartHidden }) => {
  return (
    <div className='cart-dropdown'>
      <div className='cart-items'>
      {
        cartItems.length ?
        cartItems.map(cartItem => <CartItem key={cartItem.id} item={cartItem} />)
        : 
        (<span className='cart-message' onClick={toggleCartHidden}>It's looking a bit empty here...</span>)
      }

      </div>
      <CustomButton onClick={() => {
        history.push('/checkout')
        toggleCartHidden()
      }}>Go to Checkout</CustomButton>
    </div>
  );
};

const mapStateToProps = createStructuredSelector({
  cartItems: selectCartItems
});

const mapDispatchToProps = dispatch => ({
  toggleCartHidden: () => dispatch(toggleCartHidden())
})

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(CartDropDown));