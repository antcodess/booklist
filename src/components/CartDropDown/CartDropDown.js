import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { withRouter } from 'react-router-dom';

import { selectCartItems } from '../../redux/cart/cart.selectors';
import CartItem from '../CartItem/CartItem';
import CustomButton from '../CustomButton/CustomButton';
import './CartDropDown.scss';

const CartDropDown = ({ cartItems, history }) => {
  return (
    <div className='cart-dropdown'>
      <div className='cart-items'>
      {
        cartItems.length ?
        cartItems.map(cartItem => <CartItem key={cartItem.id} item={cartItem} />)
        : 
        (<span className='cart-message'>It's looking a bit empty here...</span>)
      }

      </div>
      <CustomButton onClick={() => {
        history.push('/checkout')
      }}>Go to Checkout</CustomButton>
    </div>
  );
};

const mapStateToProps = createStructuredSelector({
  cartItems: selectCartItems
});

export default connect(mapStateToProps)(withRouter(CartDropDown));