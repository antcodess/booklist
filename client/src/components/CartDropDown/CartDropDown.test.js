import React from 'react';
import { shallow } from 'enzyme';
import { CartDropDown } from './CartDropDown';
import CartItem from '../CartItem/CartItem';
import CustomButton from '../CustomButton/CustomButton';

describe('CartDropDown Component', () => {
  let wrapper;
  let mockHistory;
  let mockCartItems;
  
  beforeEach(() => {

    mockHistory = {
      push: jest.fn()
    };

    mockCartItems = [{ id: 1}, { id: 2 }, { id: 3 }];

    const mockProps = {
      history: mockHistory,
      cartItems: mockCartItems
    };

    wrapper = shallow(<CartDropDown { ...mockProps } />);
  });

  it('should render CartDropDown Component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render empty cart message if cartItems are empty', () => {
    const mockProps = {
      history: mockHistory,
      cartItems: []
    }
    wrapper = shallow(<CartDropDown {...mockProps } />);
    expect(wrapper.exists('.cart-message')).toBe(true);
  });

  it('should render correct amount of CartItem components as cartItems length', () => {
    expect(wrapper.find(CartItem).length).toEqual(mockCartItems.length)
  });

  it('should call history.push when clicked', () => {
    wrapper.find(CustomButton).simulate('click');
    expect(mockHistory.push).toHaveBeenCalled();
  });
});