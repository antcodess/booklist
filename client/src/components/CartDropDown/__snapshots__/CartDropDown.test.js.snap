// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartDropDown Component should render CartDropDown Component 1`] = `ShallowWrapper {}`;
